x/gov/codec.proto

```diff
@@ -174,48 +160,35 @@ message Vote {
   VoteOption voted = 3;
 }
 
-// CreateTextProposalMsg creates a new governance proposal.
-message CreateTextProposalMsg {
-  weave.Metadata metadata = 1;
+// CreateProposalBase is the common info in all proposals.
+// This is separated out to easily be included in an app-specific CreateProposalMsg
+message CreateProposalBase {
```

Comment:
How about adding `Msg` -> `CreateProposalMsgBase` to make it clear that this is for the creation message?

This introduce an interesting dependency. Whenever `CreateProposalBase` is altered, all messages that embed it must increment the schema. This means that if this declaration is changed, all usages must be tracked and updated.

Reply:
It should only be used here.

But for making nice app level protobuf with one of instead of bytes for raw options, this is useful (more than manually copying fields). This is just to make helpers like in x/demo/external easier to write

Reply:
Name change is a good idea however